### ISSUE TITLE
There should be the possibility to await pending Tasks on shutdown

### ISSUE THREAD

**Issue by tibel** [tibel/Caliburn.Light#44]
When work is outsourced to an async Task it should be possible:
- wait for them on shutdown
- get a list of pending tasks

Also it should be possible to add Tasks to this list by the user.

**Comment by tibel** [comment]
Added `AsyncSubsystem` where you get notified when a new task is created.
This is the first step and can be extended later on...

Pending Tasks:
- use a `ConcurrentDictionary<int, Task>`
- the `Task.Id` is the key
- add new tasks to the dictionary
- add a task continuation that removes the task from the dictionary when it completes

When you want to wait on shutdown:
- get a copy of the dictionary `Values` collection
- use `Task.WaitAll()`

**Comment by tibel** [comment]
replace AsyncSubsystem class:
- use new Weakly version with `TaskWatched`
